phoenix-core/src/it/java/org/apache/phoenix/end2end/ViewConcurrencyAndFailureIT.java

```diff
@@ -0,0 +1,793 @@
+                    generateUniqueName());
+            String fullViewName1 = SLOW_VIEWNAME_PREFIX + "_"
+                    + generateUniqueName();
+
+            // create base table
+            String tableDdl = "CREATE TABLE " + fullTableName
+                    + "  (k INTEGER NOT NULL PRIMARY KEY, v1 INTEGER)"
+                    + tableDDLOptions;
+            stmt.execute(tableDdl);
+
+            latch1 = new CountDownLatch(1);
+            latch2 = new CountDownLatch(1);
+            ExecutorService executorService = Executors.newFixedThreadPool(
+                    1, new ThreadFactory() {
+                        @Override
+                        public Thread newThread(Runnable r) {
+                            Thread t = Executors.defaultThreadFactory()
+                                    .newThread(r);
+                            t.setDaemon(true);
+                            t.setPriority(Thread.MIN_PRIORITY);
+                            return t;
+                        }
+                    });
+
+            Future<Exception> future = executorService.submit(
+                    new CreateViewRunnable(fullTableName, fullViewName1));
+            // wait till the thread makes the rpc to add the column
+            boolean result = latch1.await(2, TimeUnit.MINUTES);
+            if (!result) {
+                fail("The create view rpc look too long");
+            }
+            tableDdl = "ALTER TABLE " + fullTableName + " DROP COLUMN v1";
+            try {
+                // drop the view WHERE condition column from the parent
+                stmt.execute(tableDdl);
+                fail("Dropping a column from a base table should fail when a"
+                        + " child view is concurrently being created whose"
+                        + " view WHERE condition depends on this column");
+            } catch (ConcurrentTableMutationException ignored) {
+            }
+            latch2.countDown();
+
+            Exception e = future.get();
+            assertNull(e);
+
+            // Now doing the same DROP COLUMN from the parent should fail with
+            // a different exception, but not a ConcurrentTableMutationException
+            // since the cell used to prevent concurrent modifications
+            // should be removed
+            try {
+                stmt.execute(tableDdl);
+                fail("Dropping a column from a parent that a child view depends"
+                        + " on should fail");
+            } catch (SQLException sqlE) {
+                assertEquals("Expected a different SQLException",
+                        CANNOT_MUTATE_TABLE.getErrorCode(),sqlE.getErrorCode());
+            }
+        }
+    }
+
+    /**
+     * Concurrently create a view which has its own new column and also try to
+     * add the same column to its parent.
+     * See PHOENIX-6191
+     */
+    @Test
+    public void testConcurrentViewCreationWithNewColParentColAddition()
```

Comment:
This is the only new test

Reply:
@ChinmaySKulkarni What is the expected behavior when it is a global view and only one (or some tenants) have the (offending) column?

Reply:
If any tenant view has the column already, then a conflicting column addition to its parent global view or even the ancestor base table will fail with a `CANNOT_MUTATE_TABLE` SQLException